dbt_meshify/linker.py

```diff
@@ -131,17 +151,68 @@ def _package_dependencies(
             },
         )
 
-        return {
+        # find the children of the shared models in the downstream project
+        package_children = [
+            {
+                'upstream_resource': project.model_relation_names[relation],
+                'downstream_resource': child,
+            }
+            for relation in relations
+            for child in other_project.manifest.child_map[project.model_relation_names[relation]]
+        ]
+
+        forward_dependencies = {
             ProjectDependency(
-                upstream=project.model_relation_names[relation],
-                downstream=other_project.model_relation_names[relation],
+                upstream_resource=child['upstream_resource'],
+                upstream_project_name=project.name,
+                downstream_resource=child['downstream_resource'],
+                downstream_project_name=other_project.name,
                 type=ProjectDependencyType.Package,
             )
-            for relation in relations
+            for child in package_children
         }
 
+        # find which models are in both manifests
+        backward_relations = self._find_relation_dependencies(
+            source_relations={
+                model.relation_name
+                for model in other_project.models.values()
+                if model.relation_name is not None
+            },
+            target_relations={
+                model.relation_name
+                for model in project.models.values()
+                if model.relation_name is not None
+            },
+        )
```

Comment:
I'm not confident that we need this `backwards_relations` calculation. Given that we're working with models for both the sources and the targets, and given that `intersection` is undirected, this should yield the same results as the `relations` call on line 141.

Is my understanding accurate?

Reply:
yes! that's absolutely correct -- we can perform the reverse operation on the same set of relations in `relations`